File: qb-engineer-ui/src/app/features/payments/models/create-payment-request.model.ts
import { CreatePaymentApplicationRequest } from './create-payment-application-request.model';

export interface CreatePaymentRequest {
  customerId: number;
  method: string;
  amount: number;
  paymentDate: string;
  referenceNumber?: string;
  notes?: string;
  applications: CreatePaymentApplicationRequest[];
}

File: qb-engineer-server/qb-engineer.core/Models/CreatePaymentRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreatePaymentRequestModel(
    int CustomerId,
    string Method,
    decimal Amount,
    DateTime PaymentDate,
    string? ReferenceNumber,
    string? Notes,
    List<CreatePaymentApplicationModel>? Applications);

public record CreatePaymentApplicationModel(
    int InvoiceId,
    decimal Amount);
